bigtable/metrics_test.go

```diff
@@ -115,14 +117,18 @@ func TestNewBuiltinMetricsTracerFactory(t *testing.T) {
 	}
 	go monitoringServer.Serve()
 	defer monitoringServer.Shutdown()
-	origExporterOpts := exporterOpts
-	exporterOpts = []option.ClientOption{
-		option.WithEndpoint(monitoringServer.Endpoint),
-		option.WithoutAuthentication(),
-		option.WithGRPCDialOption(grpc.WithTransportCredentials(insecure.NewCredentials())),
+
+	// Override exporter options
+	origCreateExporterOptions := createExporterOptions
+	createExporterOptions = func(opts ...option.ClientOption) []option.ClientOption {
+		return []option.ClientOption{
+			option.WithEndpoint(monitoringServer.Endpoint), // Connect to mock
+			option.WithoutAuthentication(),
+			option.WithGRPCDialOption(grpc.WithTransportCredentials(insecure.NewCredentials())),
+		}
```

Comment:
Manually overriding/reverting `createExporterOptions` could be very error prone in the future, especially since you can very easily forget to save the original function. Could you create a utility function in the tests to automatically save the original function then override it, like https://github.com/googleapis/google-cloud-go/blob/7e4fbc5612441c59bfaa1e5b9bbd06e3387b5c02/logging/logging_unexported_test.go#L413-L421?

Reply:
Done